tests/util.py

```diff
@@ -0,0 +1,115 @@
+import os
+import shutil
+from typing import List
+
+from dbt.main import handle_and_check
+from dbt.logger import log_manager
+from dbt.contracts.graph.manifest import Manifest
+from dbt.adapters.factory import get_adapter_by_type
+from dbt.events.test_types import IntegrationTestDebug
+from dbt.events.functions import fire_event
+
+
+# This is used in pytest tests to run dbt
+def run_dbt(args: List[str] = None, expect_pass=True):
+    # The logger will complain about already being initialized if
+    # we don't do this.
+    log_manager.reset_handlers()
+    if args is None:
+        args = ["run"]
+
+    print('Invoking dbt with {}'.format(args))
+    res, success = handle_and_check(args)
+    assert success == expect_pass, "dbt exit state did not match expected"
+    return res
+
+
+# Used in test cases to get the manifest from the partial parsing file
+def get_manifest(project_root):
+    path = project_root.join('target', 'partial_parse.msgpack')
+    if os.path.exists(path):
+        with open(path, 'rb') as fp:
+            manifest_mp = fp.read()
+        manifest: Manifest = Manifest.from_msgpack(manifest_mp)
+        return manifest
+    else:
+        return None
+
+
+def run_sql_file(sql_path, unique_schema):
```

Comment:
My sense is that we've used this to:
- Supply fixture data - before the advent of `dbt seed`, and to avoid a dependency on that feature if unrelated to the current test
- Set up database state, when we need to test a specific scenario

Is this something we want to keep around / keep using? Are there alternative patterns to avoid splitting test setup across multiple files?

Reply:
Yes, this seems to be from the really early tests. It's a good idea to look at where it's used and see if we can do something different.